src/buffer_manager.ts

```diff
@@ -622,8 +624,39 @@ export class BufferManager implements Disposable {
             this.textDocumentToBufferId.delete(document);
             unusedBuffers.push(bufId);
         });
-        unusedWindows.length && (await actions.lua("close_windows", unusedWindows));
```

Comment:
instead of a timeout parameter on `request()`, any reason to simply do the call asynchronoously? we don't use the result for anything.

    unusedWindows.length && (actions.lua("close_windows", unusedWindows).catch(e => log(e));

Reply:
I thought about this too, but I wasn't sure if it would actually be safer to handle the errors more "correctly" by keeping the editors/buffers around in the `this.textEditorToWinId` etc. maps (as I wrote in the TODO comments here). 

The other issue I thought of was that these promises might never resolve nor reject if they get stuck forever, so there could be a memory leak from just firing and forgetting them... I'm not sure how big a concern that is in practice, or if the GC is smart enough to deal with it. I guess even with `race()` this might be a problem, now that I think about it...

Reply:
Yeah, the `.catch()` approach seems simpler and clearer until we find a more robust solution.